File: canary/examples/tiny/python_ctypes/examples/probe_baseline.py
"""Baseline probe for the ctypes-based binding (dynamic FFI).

Exercises every wrapper in ``tiny_ctypes`` and checks the results
against the values an unbroken native side should produce. Exits
non-zero on the first mismatch.
"""

import sys
import tiny_ctypes as tiny


def check(name: str, expected: int, actual: int) -> None:
    if expected != actual:
        print(f"FAIL {name}: expected {expected}, got {actual}", file=sys.stderr)
        sys.exit(1)
    print(f"OK   {name} = {actual}")


def main() -> None:
    check("tiny.offset()", 42, tiny.offset())
    check("tiny.sum(2, 3)", 47, tiny.sum(2, 3))      # 2 + 3 + 42
    check("tiny.diff(5, 2)", 3, tiny.diff(5, 2))
    check("tiny.diff(2, 5)", -3, tiny.diff(2, 5))    # asymmetry matters for api_repack
    print("all checks passed")


if __name__ == "__main__":
    main()

File: canary/examples/tiny/ocaml/examples/probe_baseline.ml
(* Baseline probe: exercises every wrapper in [Tiny] and checks the
   results against the values an unbroken native side should produce.
   Exit code is non-zero on the first mismatch, so the scenario harness
   can detect Behavior-contract failures by simply running this. *)

let check name expected actual =
  if expected <> actual then begin
    Printf.eprintf "FAIL %s: expected %d, got %d\n" name expected actual;
    exit 1
  end else
    Printf.printf "OK   %s = %d\n" name actual

let () =
  check "Tiny.offset ()"      42 (Tiny.offset ());
  check "Tiny.sum 2 3"        47 (Tiny.sum 2 3);     (* 2 + 3 + 42 *)
  check "Tiny.diff 5 2"        3 (Tiny.diff 5 2);
  check "Tiny.diff 2 5"      (-3) (Tiny.diff 2 5);   (* asymmetry matters for api_repack *)
  print_endline "all checks passed"
